### PR TITLE
docs(sphinx): Add files for sphinx documentation generation

### PR DESCRIPTION
**Pull request by santiagogaray**
Hi @chriswmackey,

Sorry took me a while to get an update on this repo. I did put some time looking at your suggestion on having the option of adding the CLI section in the ``index.rst`` file if a CLI sub-package is detected. This conf.py file should be able to handle that and I hope is flexible enough to be able to be included in the ``ladybug-tools-template``. I did make some assumptions about the handling of a couple of different scenarios, described below:

If no CLI modules detected, process ends and nothing is changed.
If CLI modules detected., after creating the group files, the following can happen:
1) **cli/index.rst** file:
  a) Update an existing ``cli/index.rst`` file to add the groups, or 
  b) Create a ``cli/index.rst`` file with the groups found.
2) **doc/index.rst** file:
  a) Update an existing ``doc/index.rst`` file to add the CLI section above the API section, or
  b) Skip update if CLI section already present, or
  c) Skip update if API section not detected in file, or
  d) Skip update if no ``doc/index.rst`` file found.

I hope this makes sense Chris, happy to make any changes to the above if you have any suggestions.
Here are a couple of illustrative screenshots on how a generated CLI section and cli/index file should look like if a CLI package is found:
doc/index.rst
image
cli/index.rst
image

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+        print("[CLI doc\\index]: No index.rst file found.")
+        return -1
+    with open(index_fpath, 'r') as index_file:
+        text = index_file.read()
+
+    # Check if CLI Docs section already present
+    if re.search(r'^ *cli\/\/index *\n|^CLI Docs *\n=+\n', text, flags=re.MULTILINE):
+        return 1
+
+    print("[CLI doc\\index]: Updating index.rst file...")
```
Comment (chriswmackey) [inline docs/conf.py:514]: Ah, this is awesome!

**Review by chriswmackey — APPROVED** [review]
Thanks @santiagogaray ! This looks perfect!

And your timing is impeccable. I just made the decision 2 days ago that I will add a CLI module to this package. So we'll be able to test your code completely within the intended environment. Because I was planning to send a PR to this repo with a CLI later today, I am going to merge this now so that I can see what the docs look like without the CLI. Then, I'll request your review on the PR that I end up sending here so that we can see how the new CLI docs work.

If it all worked well, then it would be a huge help to have this new version implemented in the ladybug-tools-template.

**Comment by chriswmackey** [comment]
Actually, I realized that I'll have to send a PR first to make sure that this repo gets the docs generated as part of our CI. So I'll send that one now and request your review on it.

**Comment by santiagogaray** [comment]
I'm so glad @chriswmackey we are using this right away and put into a real test. I see you are adding some files, should I do something at this time? Happy to test out anything as soon as is ready.

**Comment by chriswmackey** [comment]
I just requested your review on a PR that is adding the CLI, @santiagogaray :
https://github.com/ladybug-tools/lbt-recipes/pull/222

Given that this is effectively a one-command CLI without any groups, though, I actually don't know if it's going to render correctly with the docs.  I also noticed that some of the commands on the other repos seem to be falling through the cracks in the docs. Like this `config` command on `honeybee-energy` that just shows you your system configuration:
https://github.com/ladybug-tools/honeybee-energy/blob/master/honeybee_energy/cli/__init__.py#L28-L56

Is there any way that we might be able to put these commands without groups into the `cli.rst` and the `cli/index.html`?

**Pull request merged**